modules/carousel/jetpack-carousel.php

```diff
@@ -540,9 +540,50 @@ class_exists( 'Jetpack_AMP_Support' )
 			 * @param array $extra_data Array of data about the site and the post.
 			 */
 			$extra_data = apply_filters( 'jp_carousel_add_data_to_container', $extra_data );
+
 			foreach ( (array) $extra_data as $data_key => $data_values ) {
-				$html = str_replace( '<div ', '<div ' . esc_attr( $data_key ) . "='" . json_encode( $data_values ) . "' ", $html );
-				$html = str_replace( '<ul class="wp-block-gallery', '<ul ' . esc_attr( $data_key ) . "='" . json_encode( $data_values ) . "' class=\"wp-block-gallery", $html );
+				// Let's grab all containers from the HTML.
+				$dom_doc = new DOMDocument();
```

Comment:
DOMDocument can be disabled, so should have a graceful fallback or bail out, e.g. https://github.com/Automattic/jetpack/blob/7.7/class.jetpack-post-images.php#L389

Reply:
👍 	fef2a1a6b02b5af974adae7302306125bc34f14c

Reply:
fef2a1a